core/src/appleMain/kotlin/Scanner.kt

```diff
@@ -35,6 +37,34 @@ public class AppleScanner internal constructor(
             .map { (cbPeripheral, rssi, advertisementData) ->
                 Advertisement(rssi.intValue, advertisementData, cbPeripheral)
             }
+
+    private fun appleOptions(): Map<Any?, *>? =
+        if (options != null) {
+            mapOf(
+                CBCentralManagerScanOptionAllowDuplicatesKey to options.allowDuplicateKeys,
+                CBCentralManagerScanOptionSolicitedServiceUUIDsKey to options.solicitedServiceUuids,
+            )
+        } else {
+            null
+        }
```

Comment:
This can be slimmed down a bit to something a bit more idiomatic like so:
```suggestion
        options?.let {
            mapOf(
                CBCentralManagerScanOptionAllowDuplicatesKey to it.allowDuplicateKeys,
                CBCentralManagerScanOptionSolicitedServiceUUIDsKey to it.solicitedServiceUuids,
            )
        }
```